web-client/csp.js

```diff
@@ -0,0 +1,24 @@
+import { getCSP, SELF, NONE, UNSAFE_INLINE, UNSAFE_EVAL } from "csp-header";
+
+export function generateCSP(isDev = false) {
+  return getCSP({
+    reportUri: isDev
+      ? ""
+      : "https://o4506303762464768.ingest.sentry.io/api/4506303812272128/security/?sentry_key=57614e75ac5f8c480aed3a2dd1528f13",
+    directives: {
+      "default-src": [SELF],
+      "script-src": isDev ? [SELF, UNSAFE_EVAL] : [SELF],
```

Comment:
I think we should investigate what causes `unsafe-eval` to be required as it significantly weakens the CSP.